script.js

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+const todoForm = document.getElementById("todo-form"),
+    todoInput = document.getElementById("todo-input"),
+    todoList = document.getElementById("todo-list");
+
+todoForm.addEventListener("submit", function (event) {
+    event.preventDefault();
+    let newTask = todoInput.value;
+
+    if (newTask === "") {
+        alert("Please, enter a task!");
+        return;
+    }
+
+    addTask(newTask);
+    todoInput.value = "";
+});
+
+function addTask(task) {
+    const listItem = document.createElement('li');
+
+    const taskText = document.createElement('span');
+    taskText.textContent = task;
+    listItem.appendChild(taskText);
+
+    const checkBox = document.createElement('input');
+    checkBox.setAttribute('type', 'checkbox');
+    listItem.appendChild(checkBox);
+
+    const deleteButton = document.createElement('button');
+    deleteButton.textContent = 'Delete';
+    listItem.appendChild(deleteButton);
+
+    const editButton = document.createElement('button');
+    editButton.textContent = 'Edit';
+    listItem.appendChild(editButton);
+
+    todoList.appendChild(listItem);
+
+    checkBox.addEventListener('change', function () {
+        if (this.checked) {
+            taskText.style.textDecoration = 'line-through';
+        } else {
+            taskText.style.textDecoration = 'none';
+        }
+    });
+
+    deleteButton.addEventListener('click', function () {
+        todoList.removeChild(listItem);
+    });
+ 
+    const input = document.createElement('input');
+    editButton.addEventListener('click', function () {
+        const isEditing = listItem.classList.contains('editing');
+
+        if (isEditing) {
```

Comment:
Please refactor to make it more readable

Reply:
Done.